From ab5825c71cb90ed6133c505608a6553b582ab72f Mon Sep 17 00:00:00 2001
From: Ismael Luceno <iluceno@iodev.co.uk>
Date: Wed, 4 Jun 2025 09:38:10 +0200
Subject: [PATCH] Revert "grub-core/genmoddep.awk: Ensure output is
 deterministic"

The code is not portable; only works with GNU AWK.

This reverts commit 87648e9c12a32bddb005b899edc44c2c9c63df82.

Origin: Source Mage
Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 grub-core/genmoddep.awk | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/grub-core/genmoddep.awk b/grub-core/genmoddep.awk
index ab457cb2b564..b61b0b9eaed9 100644
--- a/grub-core/genmoddep.awk
+++ b/grub-core/genmoddep.awk
@@ -63,9 +63,7 @@ END {
     }
     modlist = ""
     depcount[mod] = 0
-    n = asorti(uniqmods, w)
-    for (i = 1; i <= n; i++) {
-      depmod = w[i]
+    for (depmod in uniqmods) {
       modlist = modlist " " depmod;
       inverse_dependencies[depmod] = inverse_dependencies[depmod] " " mod
       depcount[mod]++
-- 
2.49.0

